cli/src/user.rs

```diff
@@ -169,78 +197,45 @@ impl User {
+                            processed_message
                         }
-                        DsMlsMessage::Plaintext(msg) => match group.verify(msg, &self.crypto) {
-                            Ok(msg) => msg,
-                            Err(e) => {
-                                log::error!(
-                                    "Error verifying MlsPlaintext: {:?} -  Dropping message.",
-                                    e
-                                );
-                                continue;
-                            }
-                        },
-                    };
-                    drop(group);
-                    let group = match groups.get_mut(&msg.group_id().to_vec()) {
-                        Some(g) => g,
-                        None => {
+                        // XXX: Should we support MlsPlaintext messages?
+                        DsMlsMessage::Plaintext(msg) => {
                             log::error!(
-                                "Error getting group {:?} for a message. Dropping message.",
-                                msg.group_id().as_slice()
+                                "Received Plaintext message: {:?} -  Dropping message.",
+                                msg
                             );
                             continue;
                         }
                     };
-                    match msg.content() {
-                        MlsPlaintextContentType::Application(application_message) => {
+
+                    // Redefine group to be the wrapper rather the MlsGroup.
+                    drop(group);
+                    let group = groups
+                        .get_mut(group_id)
+                        .expect("Failed to get group for message (after getting it earlier)!");
+
+                    match msg {
+                        ProcessedMessage::ApplicationMessage(application_message) => {
                             let application_message =
-                                String::from_utf8(application_message.as_slice().to_vec()).unwrap();
+                                String::from_utf8(application_message.message().into()).unwrap();
                             if group_name.is_none()
                                 || group_name.clone().unwrap() == group.group_name
                             {
                                 messages_out.push(application_message.clone());
                             }
                             group.conversation.add(application_message);
                         }
-                        MlsPlaintextContentType::Proposal(_proposal) => {
-                            // Store the proposal to use later when we got a
-                            // corresponding commit.
-                            group.pending_proposals.push(msg);
+                        ProcessedMessage::ProposalMessage(_proposal_ptr) => {
+                            // XXX: Do we need to do anything here?
```

Comment:
In the case where the CLI only talks to other CLIs we don't need to do anything here.